pkg/services/ngalert/schedule/schedule.go

```diff
@@ -49,6 +37,12 @@ type ScheduleService interface {
 	overrideCfg(cfg SchedulerCfg)
 }
 
+//go:generate mockery --name AlertSender --structname FakeAlertSender --inpackage --filename alertsender_mock.go --with-expecter
+type AlertSender interface {
```

Comment:
I'm not sure this is the right interface as states are not alerts. What does it mean to send a state?